src/software/firmware/srcs/activation.cpp

```diff
@@ -27,23 +27,15 @@ ActivationController::ActivationController() : m_state(STOPPED), m_timeOfLastSto
 void ActivationController::onStartButton() { m_state = RUNNING; }
 
 void ActivationController::onStopButton() {
-    switch (m_state) {
-    default:
-    case STOPPED:
-        break;
-
-    case RUNNING_READY_TO_STOP:
-        if ((millis() - m_timeOfLastStopPushed) < SECOND_STOP_MAX_DELAY_MS) {
+    if ((m_state == RUNNING_READY_TO_STOP) || (m_state == RUNNING)) {
+        if ((m_state == RUNNING_READY_TO_STOP)
+            && ((millis() - m_timeOfLastStopPushed) < SECOND_STOP_MAX_DELAY_MS)) {
```

Comment:
I think the behavior is changed here, if delta is greater than delay, we should execute "running" case code, no?

Reply:
maybe something like this then?
```suggestion
    if ((m_state == RUNNING_READY_TO_STOP)
        && ((millis() - m_timeOfLastStopPushed) < SECOND_STOP_MAX_DELAY_MS)) {
        m_state = STOPPED;
    } else if ((m_state == RUNNING_READY_TO_STOP) || (m_state == RUNNING)) {
        m_timeOfLastStopPushed = millis();
        m_state = RUNNING_READY_TO_STOP;

        Buzzer_Boot_Start();
    }
```

Reply:
Just noticed that if millis() makes a roundtrip, we can have `(millis() - m_timeOfLastStopPushed) < SECOND_STOP_MAX_DELAY_MS)` being unexpectedly true. m_state should be regularly be checked and change from RUNNING_READY_TO_STOP to RUNNING as soon as possible

Reply:
@csommelet  I don't see the behavior change you mentionned.
But I think your code is clearer than the one from @dsferruzza (sorry David :smile: )

Reply:
oups... really sorry, indeed, I did not read well sorry sorry